jme3-examples/src/main/java/jme3test/terrain/PBRTerrainAdvancedTest.java

```diff
@@ -0,0 +1,395 @@
+import com.jme3.light.AmbientLight;
+import com.jme3.light.DirectionalLight;
+import com.jme3.light.LightProbe;
+import com.jme3.material.Material;
+import com.jme3.math.ColorRGBA;
+import com.jme3.math.Vector3f;
+import com.jme3.scene.Node;
+import com.jme3.shader.VarType;
+import com.jme3.system.AppSettings;
+import com.jme3.terrain.geomipmap.TerrainQuad;
+import com.jme3.terrain.heightmap.AbstractHeightMap;
+import com.jme3.terrain.heightmap.ImageBasedHeightMap;
+import com.jme3.texture.Image;
+import com.jme3.texture.Texture;
+import com.jme3.texture.Texture.WrapMode;
+import com.jme3.texture.TextureArray;
+import java.util.ArrayList;
+import java.util.List;
+
+
+// Contains assets from CC0Textures.com, licensed under CC0 1.0 Universal.
+
+public class PBRTerrainAdvancedTest extends SimpleApplication {    
+    
+    
+    //similar to PRBTerrainTest, but this one uses texture arrays to allow for more texture slots. beware that GPUs for older devices and mobile devices may be less likely to support texture arrays
+        
+    //results look better with anti-aliasing, especially at far distances. 
+    
+    private TerrainQuad terrain;
+    private Material matTerrain;
+    private boolean triPlanar = false;
+    private float dirtScale = 24;
+    private float darkRockScale = 24;
+    private float snowScale = 64;
+    private float tileRoadScale = 64;
+    private float grassScale = 24;
+    private float marbleScale = 64;
+    private float gravelScale = 64;
+    
+    private AmbientLight ambientLight;
+    private DirectionalLight directionalLight;
+    private boolean isNight = false;
+    
+    private BitmapText keybindingsText;
+
+    public static void main(String[] args) {
+        PBRTerrainAdvancedTest app = new PBRTerrainAdvancedTest();
+        AppSettings s = new AppSettings(true);
+        s.put("FrameRate", 140);
+        
+        app.setSettings(s);
+        app.start();
+        
+ 
+       
+    }
+    
+    @Override
+    public void simpleInitApp() {
+        setupKeys();
+
+        // advanced pbr terrain matdef
+        matTerrain = new Material(assetManager, "Common/MatDefs/Terrain/AdvancedPBRTerrain.j3md");
+        
+        
+        matTerrain.setBoolean("useTriPlanarMapping", false);
+
+        // ALPHA map (for splat textures)
+        matTerrain.setTexture("AlphaMap", assetManager.loadTexture("Textures/Terrain/splat/alpha1.png"));
+        matTerrain.setTexture("AlphaMap_1", assetManager.loadTexture("Textures/Terrain/splat/alpha2.png"));
+        // this material also supports 'AlphaMap_2', so you can get up to 12 texture slots
+
+        // HEIGHTMAP image (for the terrain heightmap)
+        TextureKey hmKey = new TextureKey("Textures/Terrain/splat/mountains512.png", false);
+        Texture heightMapImage = assetManager.loadTexture(hmKey);
+
+        // load textures 
+        // it is IMPORTANT that these MUST all have the same dimensions and format in order to be put into a texture array
```

Comment:
Aren't the textures all 1K pixels on a side?

Reply:
Yes I left this comment to let the user know that the fact they share a size and format is important, since that may not be common knowledge when using texture arrays. 
Would you suggest removing the comment or changing it to something that sounds better?

Reply:
Delete the portions of the comment that say they're not currently the same size.